reactor-core/src/test/java/reactor/core/ExceptionsTest.java

```diff
@@ -156,9 +156,64 @@ public void propagateDoesntWrapRuntimeException() {
 
 	//TODO test terminate
 
+	@Test
+	void isFatalNotJvmFatalBubbling() {
+		Throwable exception = new BubblingException("expected");
+		assertThat(Exceptions.isFatal(exception))
+			.as("isFatal(bubbling)")
+			.isTrue();
+		assertThat(Exceptions.isJvmFatal(exception))
+			.as("isJvmFatal(bubbling)")
+			.isFalse();
+	}
+
+	@Test
+	void isFatalNotJvmFatalErrorCallback() {
+		Throwable exception = new ErrorCallbackNotImplemented(new IllegalStateException("expected cause"));
+		assertThat(Exceptions.isFatal(exception))
+			.as("isFatal(ErrorCallbackNotImplemented)")
+			.isTrue();
+		assertThat(Exceptions.isJvmFatal(exception))
+			.as("isJvmFatal(ErrorCallbackNotImplemented)")
+			.isFalse();
+	}
+
+	@Test
+	void isFatalAndJvmFatalVirtualMachineError() {
+		Throwable exception = new VirtualMachineError() { };
+		assertThat(Exceptions.isFatal(exception))
+			.as("isFatal(VirtualMachineError)")
+			.isTrue();
+		assertThat(Exceptions.isJvmFatal(exception))
+			.as("isJvmFatal(VirtualMachineError)")
+			.isTrue();
+	}
+
+	@Test
+	void isFatalAndJvmFatalLinkageError() {
+		Throwable exception = new LinkageError();
+		assertThat(Exceptions.isFatal(exception))
+			.as("isFatal(LinkageError)")
+			.isTrue();
+		assertThat(Exceptions.isJvmFatal(exception))
+			.as("isJvmFatal(LinkageError)")
+			.isTrue();
+	}
+
+	@Test
+	void isFatalAndJvmFatalThreadDeath() {
+		Throwable exception = new ThreadDeath();
+		assertThat(Exceptions.isFatal(exception))
+			.as("isFatal(ThreadDeath)")
+			.isTrue();
+		assertThat(Exceptions.isJvmFatal(exception))
+			.as("isJvmFatal(ThreadDeath)")
+			.isTrue();
+	}
+
 	@Test
 	public void throwIfFatalThrowsBubbling() {
-		BubblingException expected = new BubblingException("expected");
+		BubblingException expected = new BubblingException("expected to be logged");
```

Comment:
Did you consider using a test appender to actually validate it's logged?

Reply:
yes, that's a little more work but fortunately all the `Logger` instances in tests can copy their output to a `TestLogger`. will do that shortly.